core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -36,12 +36,16 @@ oppia.directive('collectionEditorNavbar', [function() {
           EVENT_UNDO_REDO_SERVICE_CHANGE_APPLIED) {
         $scope.collectionId = GLOBALS.collectionId;
         $scope.collection = CollectionEditorStateService.getCollection();
-        $scope.isPrivate = GLOBALS.isPrivate;
-        $scope.canUnpublish = GLOBALS.canUnpublish;
         $scope.validationIssues = [];
         $scope.isSaveInProgress = (
           CollectionEditorStateService.isSavingCollection);
 
+        CollectionRightsBackendApiService.getCollectionRights(
+          $scope.collectionId).then(function(response) {
+          $scope.isPrivate = response.is_private;
```

Comment:
I think you might need to be more careful at the outset. There's a short period after the page loads where the rights status is undefined. Perhaps you can have a variable rightsHasLoaded that's initially false, and when it's false, don't display any of the save/publish buttons in the frontend (to avoid a brief period where you might be displaying the wrong thing)? Similar considerations apply elsewhere.